includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -176,7 +176,7 @@ public function capture_terminal_payment( $request ) {
 			// Store receipt generation URL for mobile applications in order meta-data.
 			$order->add_meta_data( 'receipt_url', get_rest_url( null, sprintf( '%s/payments/readers/receipts/%s', $this->namespace, $intent->get_id() ) ) );
 			// Actualize order status.
-			$this->order_service->mark_terminal_payment_completed( $order, $intent_id, $intent_status, $charge_id );
+			$this->order_service->mark_terminal_payment_completed( $order, $intent_id, $result['status'], $charge_id );
```

Comment:
@jessepearson @kalessil  Do we need this method call when we have a similar one in the `WC_Payment_Gateway_WCPay::capture_charge` method?

https://github.com/Automattic/woocommerce-payments/blob/cbf2927962e26b81d8b0e245678222254cc993a7/includes/class-wc-payment-gateway-wcpay.php#L2091-L2093

Reply:
IIRC, we need it here due to `$result = $is_intent_captured ? $result_for_captured_intent : $this->gateway->capture_charge( $order, false );`.

Reply:
Right, that makes perfect sense :)